velox/exec/HashBuild.cpp

```diff
@@ -815,11 +908,18 @@ bool HashBuild::finishHashBuild() {
               spillStats);
         };
   }
+
+  // For hash table caching: the last driver caches the merged table.
+  std::shared_ptr<BaseHashTable> table = std::move(table_);
+  if (useHashTableCache()) {
+    setHashTableInCache(table);
```

Comment:
Let's check useHashTableCache() inside setHashTableInCache? only set if it is true?